grakn-engine/src/main/java/ai/grakn/engine/cache/EngineCacheStandAlone.java

```diff
@@ -39,59 +36,40 @@
  *
  *    We cannot relay on {@link ai.grakn.concept.ConceptId}s because the indexed lookup maybe faulty with
  *    vertices in need of post processing.
+ *
+ *    This stand alone version keeps everything in memory when running a single engine.
  * </p>
  *
  * @author fppt
  */
-public class EngineCache implements ConceptCache{
+//TODO: Maybe we can merge this with distributed engine cache using Kafka for example?
+public class EngineCacheStandAlone extends EngineCacheAbstract{
     //These are maps of keyspaces to indices to vertex ids
     private final Map<String, Map<String, Set<ConceptId>>> castings;
     private final Map<String, Map<String, Set<ConceptId>>> resources;
 
-    private final AtomicBoolean saveInProgress;
-    private static EngineCache instance=null;
-    private final AtomicLong lastTimeModified;
+    private static EngineCacheStandAlone instance=null;
 
-    public static synchronized EngineCache getInstance(){
-        if(instance==null) instance=new EngineCache();
+    public static synchronized EngineCacheStandAlone getCache(){
+        if(instance == null) instance = new EngineCacheStandAlone();
```

Comment:
We can probably eliminate this singleton as well, like `EngineCacheDistributed`.